### PR TITLE
migrate more modules to be multi-thread safe

### PR DESCRIPTION
**Pull request by fwyzard**
- HLTEcalPhiSymFilter: migrate to global::EDFilter
- ESRecHitProducer: migrate to stream::EDProducer

### CODE REVIEW

**Comment by fwyzard** [comment]
@cmsbuild please test

**Comment by fwyzard** [comment]
@cmsbuild please test

File: CommonTools/UtilAlgos/interface/EventSelectorAdapter.h [hunk 1]
```diff
@@ -31,7 +31,7 @@ class EventSelectorAdapter : public edm::EDFilter
   virtual ~EventSelectorAdapter() {}
 
  private:
-  bool filter(edm::Event& evt, const edm::EventSetup& es) override { return eventSelector_(evt, es); }
+  bool filter(edm::StreamID, edm::Event& evt, const edm::EventSetup& es) const override { return eventSelector_(evt, es); }
```
Comment (Dr15Jones) [inline CommonTools/UtilAlgos/interface/EventSelectorAdapter.h]: I highly advise to add ```override``` on all such interfaces to guarantee one doesn't make a typo.

Comment (fwyzard) [inline CommonTools/UtilAlgos/interface/EventSelectorAdapter.h]: hi Chris, I do agree, but I'm not actually going to fix all the CommonTools modules...

**Comment by fwyzard** [comment]
@cmsbuild please test

**Comment by fwyzard** [comment]
+1

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/8765	 e6c717e
- changes in reco-related parts are next to trivial
- jenkins comparisons show no differences in DQM or fwlite comparisons

**Pull request merged**